torch/quantization/convert_modules.py

```diff
@@ -0,0 +1,221 @@
+    for i in range(1, len(splits)):
+        mod = mod._modules[splits[i]]
+    return mod
+
+def addQuantDeQuant(modInstance, qConfigDict):
+    r"""Add AbstractQuant and AbstractDeQuant module and modify forward function
+    to call quant dequant accordign to qConfigDict
+    """
+    mod_key_list = []
+    for name, mod in modInstance.named_modules():
+        dict_key = get_config_key(name, qConfigDict)
+        if dict_key is not None:
+            mod_key_list.append(dict_key)
+
+    for name in set(mod_key_list):
+        if name is not '':
+            addQuantDeQuantModule(getModule(modInstance, name), qConfigDict[name])
+        else:
+            addQuantDeQuantModule(modInstance, qConfigDict[name])
+
+def prepare(modInstance, qConfigDict):
+    r"""Transform a module to a quantized module according to qConfigDict,
+    it adds observer and abstract quant dequant modules to the module. and
+    changes the forward method to invoke them
+
+    Args:
+        modInstance: instance of the module we want to transform
+        qCofnigDict: dictionary that maps from name of submodule to quantization
+                     configuration
+    """
+    addObserver(modInstance, qConfigDict)
+    addQuantDeQuant(modInstance, qConfigDict)
+
+class AbstractQuant(nn.Module):
+    r"""Abstract quantize module which will be replace to actual Quantize module
+    in swapModule function
+    """
+    def __init__(self, qconfig):
+        super(AbstractQuant, self).__init__()
+        self.add_module('observer', qconfig.activation())
+
+    def forward(self, x):
+        self.observer(x)
+        return x
+
+class AbstractDeQuant(nn.Module):
+    r"""Abstract dequantize module which will be replace to actual DeQuantize module
+    in swapModule function
+    """
+    def __init__(self):
+        super(AbstractDeQuant, self).__init__()
+
+    def forward(self, x):
+        return x
+
+def calculateQParams(MyModule):
+    r""" Calculates Quantizer parameters for activation tensors based on observer statistics
+    Updates the qparams parameter for each module with the quantizer params.
+
+    Args:
+        MyModule: Model with observer stats
+    """
+    if hasattr(MyModule, 'observer'):
+        if isinstance(MyModule.observer, Observer):
+            # Simple symmetric quantization over entire observed range
+            MyModule.qparams = MyModule.observer.calculate_qparams()
+    for Module in MyModule.children():
+        calculateQParams(Module)
+
+def quantize(module, qConfigDict, eval_fn, *eval_args):
+    r"""Given a float module and qConfigDict, convert it to a quantized module
+
+    First it will prepare the module to add observer and abstract quant
+    deqaunt modules, then it calls `eval_fn` which will run the calibration
+    step and observers will record the stats of the tensors, after that we will
+    call `convert` which will convert the module to a quantized module.
+    """
+    prepare(module, qConfigDict)
```

Comment:
I wonder whether `quantize()` overall is even needed. It might be fine to just ask user to explicitly call prepare/convert. I'd rename `convert` into something different then

Reply:
don't we need to have eval_fn as well?
I think it's just a wrapper over these three functions listed below and it gets us a one-line API for quantizing a model.